### PR TITLE
fix azure disk storage account init issue

### LINKED ISSUE #50883: Azure data disk should provision storage account on on-demand
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
The azure_dd plugin will always provision two instances of storage account with different type during init.

There could be cases when:
1. User may only want to use one certain type of storage account, then the other type of storage account would be not in use.
2. User chooses to use managed disk, both of the storage account would be not in use. 

There is a limit for storage account numbers per subscription, and the storage account should be created with a on-demand policy.

Log:
```
I0818 03:00:14.709321       1 azure_blobDiskController.go:354] azureDisk - BlobDiskController init process  will create new storageAccount:pvc2365976634001 type:Premium_LRS
I0818 03:00:14.709693       1 azure_blobDiskController.go:354] azureDisk - BlobDiskController init process  will create new storageAccount:pvc2365976634002 type:Standard_LRS
I0818 03:00:14.739106       1 azure_blobDiskController.go:602] azureDisk - Creating storage account pvc2365976634001 type Premium_LRS 
I0818 03:00:14.759665       1 azure_blobDiskController.go:602] azureDisk - Creating storage account pvc2365976634002 type Standard_LRS
```
Code:
https://github.com/kubernetes/kubernetes/blob/9e97b5249bdd105705bfa6a7c6da477e8bde57d1/pkg/cloudprovider/providers/azure/azure_blobDiskController.go#L335-L366

**What you expected to happen**:
The storage accounts be created on-demand.

**How to reproduce it (as minimally and precisely as possible)**:
Setup a kubernetes v1.7.2 cluster on Azure, create a dynamic pvc.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T23:15:59Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.2", GitCommit:"922a86cfcd65915a9b2f69f3f193b8907d741d9c", GitTreeState:"clean", BuildDate:"2017-07-21T08:08:00Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration**:
Azure

### LINKED ISSUE #54570: wrong controller-master detection
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG
/kind bug

**What happened**:
[This line](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_blobDiskController.go#L546) is buggy. 
```go
func (c *BlobDiskController) shouldInit() bool {
	if os.Args[0] == "kube-controller-manager" || (os.Args[0] == "/hyperkube" && os.Args[1] == "controller-manager") {
		swapped := atomic.CompareAndSwapInt64(&initFlag, 0, 1)
		if swapped {
			return true
		}
	}
	return false
}
```
It prevents running hyperkube from other path. In fact, it shouldn't assume the code is just running in controller master at all, especially when cloud provider is running out of tree.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
@kubernetes/sig-windows-bugs 
@andyzhangx 
**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### LINKED ISSUE #55776: Azure disk dose not work as expected
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
For unmanaged disk, now it is not working
**What you expected to happen**:
Works well
**How to reproduce it (as minimally and precisely as possible)**:
```
# kubectl get sc
NAME      PROVISIONER                AGE
scazure   kubernetes.io/azure-disk   4h
# kubectl get sc scazure -o yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: 2017-11-15T03:15:24Z
  name: scazure
  resourceVersion: "5718"
  selfLink: /apis/storage.k8s.io/v1/storageclasses/scazure
  uid: 3896d86f-c9b3-11e7-9c74-000d3a139483
parameters:
  location: eastus
  storageAccount: storage
provisioner: kubernetes.io/azure-disk
reclaimPolicy: Delete

# kubectl describe pvc azpvc
Name:          azpvc
Namespace:     default
StorageClass:  scazure
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-class=scazure
               volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/azure-disk
Capacity:      
Access Modes:  
Events:
  Type     Reason              Age               From                         Message
  ----     ------              ----              ----                         -------
  Warning  ProvisioningFailed  6s (x2 over 17s)  persistentvolume-controller  Failed to provision volume with StorageClass "scazure": Create Storage Account: %!!(MISSING)(EXTRA string=001), error: storage.AccountsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidDoubleEncodedRequestUri" Message="The request URI 'https://management.azure.com:443/subscriptions/.../resourceGroups/openshift/providers/Microsoft.Storage/storageAccounts/%!!(MISSING)(EXTRA%!s(MISSING)tring=001)?api-version=2017-06-01' is not valid, because it contains double encoding sequence '%!'(MISSING)."
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
# kubectl version
Client Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.0-alpha.2.734+2d64ce5e8e45e2", GitCommit:"2d64ce5e8e45e26b02492d2b6c85e5ebfb1e4761", GitTreeState:"clean", BuildDate:"2017-11-14T03:26:54Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.0-alpha.2.734+2d64ce5e8e45e2", GitCommit:"2d64ce5e8e45e26b02492d2b6c85e5ebfb1e4761", GitTreeState:"clean", BuildDate:"2017-11-14T03:26:54Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): RHEL7
- Kernel (e.g. `uname -a`):
- Install tools: ansible
- Others:

### PR DESCRIPTION
**Pull request by andyzhangx**
**What this PR does / why we need it**:
There are two issues for the original azure disk storage account initialaztion code:
1) wrong controller-master detection, see issue #54570, #55776 
2) should not initialize two storage account even if it's not necessary, see issue #50883

This PR would fix the above two issues:
For 1: remove the controller-master process binding
For 2: remove the storage account initialization process, just create on demand

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #54570
Fixes #55776
Fixes #50883

**Special notes for your reviewer**:
@rootfs @karataliu 

**Release note**:

```
fix azure disk storage account init issue
```
/sig azure

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_blobDiskController.go [hunk 1]
```diff
@@ -330,41 +325,6 @@ func (c *BlobDiskController) init() error {
 	}
 	c.accounts = accounts
 
-	if len(c.accounts) == 0 {
```
Comment (karataliu) [inline pkg/cloudprovider/providers/azure/azure_blobDiskController.go]: Also update comment lines before init method

File: pkg/cloudprovider/providers/azure/azure_blobDiskController.go [hunk 2]
```diff
@@ -543,13 +503,7 @@ func (c *BlobDiskController) getDiskCount(SAName string) (int, error) {
 // and we only do that in the controller
 
 func (c *BlobDiskController) shouldInit() bool {
```
Comment (karataliu) [inline pkg/cloudprovider/providers/azure/azure_blobDiskController.go]: not needed?

File: pkg/cloudprovider/providers/azure/azure_blobDiskController.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_blobDiskController.go]: fixed

File: pkg/cloudprovider/providers/azure/azure_blobDiskController.go (hunk 2 above)
Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_blobDiskController.go]: init() func removed

**Pull request merged**